clusterloader2/pkg/measurement/common/slos/windows_node_resource_usage.go

```diff
@@ -68,20 +74,48 @@ func convertToPerfData(samples []*model.Sample) *measurementutil.PerfData {
 	return perfData
 }
 
-// Gather gathers the metrics and convert to json summary
-func (w *windowsResourceUsageGatherer) Gather(executor QueryExecutor, startTime time.Time, config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
-	samples, err := executor.Query(cpuUsageQueryTop10, time.Now())
+func convertToMemoryPerfData(samples []*model.Sample) *measurementutil.PerfData {
+	perfData := &measurementutil.PerfData{Version: currentWindowsResourceUsageMetricsVersion}
+	for _, sample := range samples {
+		item := measurementutil.DataItem{
+			Data: map[string]float64{
+				"Memory_Usage": math.Round(float64(sample.Value)*100/(1024*1024)) / 100,
+			},
+			Unit: "MB",
+			Labels: map[string]string{
+				"Process": string(sample.Metric["process"]),
+			},
+		}
+		perfData.DataItems = append(perfData.DataItems, item)
+	}
+	return perfData
+}
+
+func getSummary(query string, converter func(samples []*model.Sample) *measurementutil.PerfData, metricsName string, executor QueryExecutor, startTime time.Time, config *measurement.MeasurementConfig) (measurement.Summary, error) {
```

Comment:
Two nits:
1. please define a type for the function type, e.g.:
```
type convertFunc func([]*model.Sample) *measurementutil.PerfData
```
2. startTime seems unused - please remove

Reply:
Thanks, fixed